### ISSUE TITLE
Explore ways of generalizing Settings and implement

### ISSUE THREAD

**Issue by mj3cheun** [uncharted-aske/HMI#128]
The below is copy and pasted from slack for context:

For the settings component the reason why it wasnt generalized (i tried it and i didnt like the results so i reverted the changes) is that if we do that then all the buttons in addition to all the classes will be relocated from the settings component into the parent class, such as bio or knowledge etc. because of the type of props that need to be attached to the buttons 

basically generalizing the component was more complex than i thought it would be

i can give it another go tho, however the only reasonable solution i think is to pass in an array of buttons to generate with ids and the properties said buttons should have
